centreon/www/front_src/src/Resources/Graph/Performance/TimePeriods/PopoverCustomTimePeriodPicker.tsx

```diff
@@ -0,0 +1,252 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+
+import dayjs from 'dayjs';
+import { useAtomValue } from 'jotai/utils';
+import { and, cond, equals } from 'ramda';
+import { useTranslation } from 'react-i18next';
+import { makeStyles } from 'tss-react/mui';
+
+import {
+  FormHelperText,
+  Popover,
+  PopoverOrigin,
+  PopoverReference,
+  Typography
+} from '@mui/material';
+import { LocalizationProvider } from '@mui/x-date-pickers';
+
+import { userAtom } from '@centreon/ui-context';
+
+import {
+  CustomTimePeriod,
+  CustomTimePeriodProperty
+} from '../../../Details/tabs/Graph/models';
+import {
+  labelEndDate,
+  labelEndDateGreaterThanStartDate,
+  labelFrom,
+  labelStartDate,
+  labelTo
+} from '../../../translatedLabels';
+import useDateTimePickerAdapter from '../../../useDateTimePickerAdapter';
+
+import DateTimePickerInput from './DateTimePickerInput';
+import { AnchorReference } from './models';
+
+const useStyles = makeStyles()((theme) => ({
+  error: {
+    textAlign: 'center'
+  },
+  paper: {
+    '& .MuiPopover-paper': {
+      minWidth: 250,
+      padding: theme.spacing(1)
+    }
+  }
+}));
+
+interface AcceptDateProps {
+  date: Date;
+  property: CustomTimePeriodProperty;
+}
+
+interface Props {
+  acceptDate: (props: AcceptDateProps) => void;
+  anchorOrigin?: PopoverOrigin;
+  anchorReference?: PopoverReference;
+  classNameError?: string;
+  classNamePaper?: string;
+  classNamePicker?: string;
+  customTimePeriod: CustomTimePeriod;
+  disabledPickerEndInput?: boolean;
+  disabledPickerStartInput?: boolean;
+  getIsErrorDatePicker?: (value: boolean) => void;
+  maxDatePickerEndInput?: Date | dayjs.Dayjs;
+  maxDatePickerStartInput?: Date;
+  minDatePickerEndInput?: Date;
+  minDatePickerStartInput?: Date;
+  onClose: () => void;
+  onCloseEndPicker?: (isClosed: boolean) => void;
+  onCloseStartPicker?: (isClosed: boolean) => void;
+  open: boolean;
+  pickerEndWithoutInitialValue?: boolean;
+  pickerStartWithoutInitialValue?: boolean;
+  reference?: AnchorReference;
+  renderBody?: JSX.Element;
+  renderFooter?: JSX.Element;
+  renderTitle?: JSX.Element;
+  setPickerEndWithoutInitialValue?: Dispatch<SetStateAction<boolean>>;
+  setPickerStartWithoutInitialValue?: Dispatch<SetStateAction<boolean>>;
+  transformOrigin?: PopoverOrigin;
+}
+
+const PopoverCustomTimePeriodPickers = ({
+  reference,
+  anchorReference = 'none',
+  anchorOrigin = {
+    horizontal: 'center',
+    vertical: 'top'
+  },
+  transformOrigin = {
+    horizontal: 'center',
+    vertical: 'top'
+  },
+  open,
+  classNamePaper,
+  classNamePicker,
+  customTimePeriod,
+  acceptDate,
+  renderTitle,
+  renderBody,
+  renderFooter,
+  pickerStartWithoutInitialValue,
+  pickerEndWithoutInitialValue,
+  setPickerStartWithoutInitialValue,
+  setPickerEndWithoutInitialValue,
```

Comment:
Instead of using those props and the picker input to `null` if `withoutInitialValue` is true, can you directly set de picker's value (I mean `customTimePeriod.start`and `customTimePeriod.end`) to `null` then?